pytorch_lightning/accelerators/accelerator_connector.py

```diff
@@ -0,0 +1,472 @@
+            use_ddp_cpu_slurm = use_ddp_cpu_spawn and self.is_slurm_managing_tasks
+            # use_ddp_sharded = self.distributed_backend == "ddp_sharded"
+            # use_ddp_sharded_spawn = self.distributed_backend == "ddp_sharded_spawn"
+
+            if self.on_tpu:
+                ddp_plugin_cls = TPUSpawnPlugin
+
+            # ddp script mode uses the same flags as TE
+            # TODO: decouple from TE
+            if os.environ.get("PL_IN_DDP_SUBPROCESS", False):
+                use_torchelastic_ddp = False
+
+            # fixme
+            # if use_ddp_sharded:
+            #     ddp_plugin_cls = DDPShardedPlugin
+            # elif use_ddp_sharded_spawn:
+            #     ddp_plugin_cls = DDPSpawnShardedPlugin
+            if use_ddp_cpu_slurm or use_slurm_ddp or use_ddp_cpu_torch_elastic or use_torchelastic_ddp:
+                ddp_plugin_cls = DDPPlugin
+            elif use_ddp_spawn or use_ddp_cpu_spawn:
+                ddp_plugin_cls = DDPSpawnPlugin
+            else:
+                ddp_plugin_cls = DDPPlugin
+
+            plugin = ddp_plugin_cls(
+                parallel_devices=self.parallel_devices,
+                num_nodes=self.num_nodes,
+                cluster_environment=cluster_environment,
+                sync_batchnorm=self.sync_batchnorm,
+            )
+        elif self.use_dp:
+            plugin = DataParallelPlugin(parallel_devices=self.parallel_devices)
+        elif self.use_horovod:
+            plugin = HorovodPlugin(parallel_devices=self.parallel_devices)
+        elif self.on_tpu:
+            plugin = SingleTPUPlugin(self.tpu_id)
+        else:
+            plugin = SingleDevicePlugin(device=torch.device(f"cuda:{self.root_gpu}" if self.on_gpu else "cpu"))
+        return plugin
+
+    def select_accelerator(self):
+        if isinstance(self.distributed_backend, Accelerator):
+            # custom accelerator from user
+            return self.distributed_backend
+
+        if self.on_gpu:
+            acc_cls = GPUAccelerator
+        elif self.on_tpu:
+            acc_cls = TPUAccelerator
+        else:
+            acc_cls = CPUAccelerator
+
+        return acc_cls(
+            precision_plugin=self.select_precision_plugin(),
+            training_type_plugin=self.select_training_type_plugin(),
+        )
+
+    def select_cluster_environment(self):
+        if self.cluster_environment is not None:
+            return self.cluster_environment
+        if self.is_slurm_managing_tasks:
+            env = SLURMEnvironment()
+        elif self.is_using_torchelastic:
+            env = TorchElasticEnvironment()
+            # TODO: decouple DDP from TE
+            #   maybe introduce a DefaultEnvironment?
+            os.environ["PL_IN_DDP_SUBPROCESS"] = "1"
+        else:
+            # TODO: maybe introduce a DefaultEnvironment?
+            env = TorchElasticEnvironment()
+        return env
+
+    def set_distributed_mode(self):
```

Comment:
This function should be cleaned out in another PR. It uses old logic.

Reply:
I merged all changes from feature/dev1.2 branch here this morning, this is the logic we currently use to select plugins based on the trainer args.